### PR TITLE
 Adopt new voting queue UI - Closes #4466

### LINKED ISSUE #4466: Adopt my voting queue UI changes according to new design
### Description

- Implement new [design](https://www.figma.com/file/KcrDpvWEKQhdGwNd4CZ5NY/Desktop-Prototype?node-id=38%3A44988) changes
- Vote to delegate page

### Acceptance Criteria
- Update voting queue icon from top bar
- Users should be able to review voting queue
- User should be able to see send vote
- Users should be able to preview delegate summary before add to queue
- Users should be able to review/edit voting queue before submitting transaction to network
- Make sure transaction working fine
- Add/Update required unit tests

### PR DESCRIPTION
**Pull request by eniolam1000752**
### What was the problem?

This PR resolves #4466

### How was it solved?

- [x] Refactor voting redux middleware, reducers and actions
- [x] Ensure edit and add vote flow works with the voting queue
- [x] Refactor UI for voting queue form
- [x] Refactor UI for voting queue summary
- [x] Implement voting confirmation modal
- [x] update unit tests

### How was it tested?

visually
jest

### CODE REVIEW

**Comment by reyraa** [comment]
A few issues here:
 - In `Vote amount ()`, what is planned to be inside the parenthesis? 
 - The token symbol should be displayed next to the balance. Also, "Available Bal"? It's not readable. Let' write the entire word. If needed, lets expand the modal by an extra 10px to open some room for the missing letters.
 - If the exchange ratio is not available, we should drop the exchange value. Showing `0.00USD` doesn't seem correct. Also, some times the exchange values shows up, some time not.
 - Although the available balance is `0.85`,I'm able to input `10`. Which seems invalid.
 
Screenshot 2022-10-18 at 13 12 03


Now when I click continue voting, instead of simply closing the modal, and allowing me to visit another validator and vote, it opens the edit vote form which has the following issues
- The vote value is empty. Considering I just voted to this delegate, I expect to see the same value.
- The form is invalid. I haven't interacted with the form yet, it should be simply empty, not invalid.

Screenshot 2022-10-18 at 13 14 27

File: src/modules/dpos/validator/api/index.js [hunk 1]
```diff
@@ -3,10 +3,12 @@ import http from 'src/utils/api/http';
 import { HTTP_PREFIX } from 'src/const/httpCodes';
 import ws, { subscribe, unsubscribe } from 'src/utils/api/ws';
 import { extractAddressFromPublicKey } from '@wallet/utils/account';
+import client from 'src/utils/api/client';
+import { API_VERSION } from 'src/const/config';
 
 export const httpPaths = {
   delegates: `${HTTP_PREFIX}/accounts`,
-  votesSent: `${HTTP_PREFIX}/votes_sent`,
+  votesSent: `/api/${API_VERSION}/dpos/votes/sent`,
```
Comment (reyraa) [inline src/modules/dpos/validator/api/index.js]: Either we refactor all endpoints to look like this or change the `HTTP_PREFIX` value. This causes inconsistency, hence increasing the likelihood of regressions.

File: src/modules/dpos/validator/api/index.js [hunk 2]
```diff
@@ -37,40 +39,40 @@ const getDelegateProps = ({ address, publicKey, username }) => {
...
-  network,
-  baseUrl,
-});
+export const getDelegate = ({ params = {}, network, baseUrl }) =>
+  http({
+    path: httpPaths.delegates,
+    params: { ...getDelegateProps(params), isDelegate: true },
+    network,
+    baseUrl,
+  });
```
Comment (reyraa) [inline src/modules/dpos/validator/api/index.js:47]: In general, please refrain from reformatting the code which you do not refactor. This will cause unnecessary conflicts, increases the number of files you changed by your PR, making the review a longer task. We'll change it when we need to change it.

File: src/modules/dpos/validator/api/index.js [hunk 3]
```diff
@@ -140,16 +138,12 @@ export const getDelegates = ({
...
-}) => http({
-  path: httpPaths.votesSent,
-  params: getDelegateProps({ address: params.address, publicKey: params.publicKey }),
-  network,
-  baseUrl,
-});
+export const getVotes = ({ params = {} }) =>
+  client.rest({
+    url: httpPaths.votesSent,
+    method: 'get',
```
Comment (reyraa) [inline src/modules/dpos/validator/api/index.js]: Isn't `GET` the default method? Why do you need to pass it?

File: src/modules/dpos/validator/api/index.test.js [hunk 4]
```diff
@@ -101,7 +109,9 @@ describe('API: LSK Delegates', () => {
       await delegate.getDelegates({
         network,
         params: {
-          addressList, limit: 5, offset: 3,
+          addressList,
+          limit: 5,
+          offset: 3,
```
Comment (reyraa) [inline src/modules/dpos/validator/api/index.test.js:114]: As explained above, reformatting a code that you do not intend to refactor is a bad habit. You should refrain. But Additionally, I don't get how your formater is configured. In some places, you have put objects with up to 3 members in a single line, and in some cases you have broken  them down into multiple lines. This is inconsistent. 
I'd recommend you revert all of them.

File: src/modules/dpos/validator/components/EditVote/index.js [hunk 5]
```diff
@@ -209,8 +212,8 @@ const EditVote = ({ history, voteEdited, network, voting }) => {
             </>
           )}
         </BoxContent>
-        <BoxFooter direction={mode === 'edit' ? 'horizontal' : 'vertical'}>
-          {mode === 'edit' && (
+        <BoxFooter direction={mode === 'edit' && isForm ? 'horizontal' : 'vertical'}>
```
Comment (reyraa) [inline src/modules/dpos/validator/components/EditVote/index.js:210]: retain either `mode`, or `isForm`.  they appear to perform a similar task.

File: src/utils/api/http.js [hunk 6]
```diff
@@ -18,8 +18,10 @@ const http = ({
 }) => {
   try {
     // @todo remove the optional chain
+    
     const url = new URL(baseUrl ? `${baseUrl}${path}`
-      : `${network?.networks?.LSK?.serviceUrl ?? 'https://testnet-service.lisk.com'}${path}`);
+    // @Todo: this should be reverted when stable test service depoyment has been done
+      : `${network?.networks?.LSK?.serviceUrl ?? 'http://165.22.29.229:9901'}${path}`);
```
Comment (reyraa) [inline src/utils/api/http.js]: This is not to be pushed. Please revert.

**Review by reyraa — CHANGES_REQUESTED** [review]

File: src/modules/dpos/validator/components/VoteStatus/Status.js [hunk 7]
```diff
@@ -1,33 +1,28 @@
...
-      </header>
-      <TxBroadcaster
-        title={template.title}
-        illustration="vote"
-        status={status}
-        message={template.message}
-      />
-    </Box>
+    <div className={styles.container}>
+      {status.code === 'BROADCAST_SUCCESS' ? (
```
Comment (reyraa) [inline src/modules/dpos/validator/components/VoteStatus/Status.js]: Please use the enum/constant.

File: src/modules/dpos/validator/components/VoteSuccessfulModal/VoteSuccessfulModal.css [hunk 8]
```diff
@@ -0,0 +1,50 @@
+.successContainer {
+  display: 'flex';
+  flex-direction: column;
+  align-items: center;
+
+  & header {
+    @mixin headingNormal very-bold;
+
+    justify-content: center !important;
+    font-size: var(--font-size-h3);
```
Comment (reyraa) [inline src/modules/dpos/validator/components/VoteSuccessfulModal/VoteSuccessfulModal.css]: Use a different mixin instead

File: src/modules/dpos/validator/components/VoteSuccessfulModal/VoteSuccessfulModal.css [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    justify-content: center;
+
+    & .illustartion {
+      width: 90px;
+    }
+
+    & .illustartion ~ h4 {
+      @mixin contentLargest semi-bold;
+
+      font-size: var(--font-size-h6);
```
Comment (reyraa) [inline src/modules/dpos/validator/components/VoteSuccessfulModal/VoteSuccessfulModal.css]: Use a different mixin instead

File: src/modules/dpos/validator/components/VoteSuccessfulModal/VoteSuccessfulModal.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+import BoxContent from 'src/theme/box/content';
+import BoxFooter from 'src/theme/box/footer';
+import Illustration from 'src/modules/common/components/illustration';
+import styles from './VoteSuccessfulModal.css';
+
+function VoteSuccessfulModal({ history, statusMessage }) {
+  const { t } = useTranslation();
+
+  const handleBackToDelegate = useCallback(() => {
+    history.push('/delegates');
```
Comment (reyraa) [inline src/modules/dpos/validator/components/VoteSuccessfulModal/VoteSuccessfulModal.js]: Should this come from the routes file?

File: src/modules/dpos/validator/components/VoteSuccessfulModal/VoteSuccessfulModal.js [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+  }, [history]);
+
+  return (
+    <Box className={styles.successContainer}>
+      <BoxHeader>Voting confirmation</BoxHeader>
```
Comment (reyraa) [inline src/modules/dpos/validator/components/VoteSuccessfulModal/VoteSuccessfulModal.js]: i18n

File: src/modules/dpos/validator/components/VoteSuccessfulModal/VoteSuccessfulModal.js [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+      <BoxContent>
+        <Illustration className={styles.illustartion} name="votingSuccess" />
+        <h4>Vote(s) has been submitted</h4>
```
Comment (reyraa) [inline src/modules/dpos/validator/components/VoteSuccessfulModal/VoteSuccessfulModal.js]: i18n

File: src/modules/dpos/validator/components/VotingQueue/index.js [hunk 13]
```diff
@@ -10,6 +11,14 @@ import Status from '../VoteStatus';
 import styles from './styles.css';
 
 const VotingQueue = ({ history, processLaunchProtocol }) => {
+  const [{ step }, setMultiStepState] = useState({});
+
+// @Todo this is just a place holder pending when dpos constants are integrated by useDposContants hook
```
Comment (reyraa) [inline src/modules/dpos/validator/components/VotingQueue/index.js]: Is there a ticket number?

File: src/modules/dpos/validator/components/delegateProfile/delegateVoteButton.js [hunk 14]
```diff
@@ -5,24 +5,24 @@ import DialogLink from 'src/theme/dialog/link';
...
   }, [sentVotes]);
 
   return (
     <DialogLink component="editVote">
       {hasSentVoteToDelegate ? (
-        <SecondaryButton disabled={sentVotesLoading || isBanned}>{t('Edit vote')}</SecondaryButton>
+        <SecondaryButton disabled={sentVotesLoading || isBanned || !currentAddress}>{t('Edit vote')}</SecondaryButton>
       ) : (
-        <PrimaryButton disabled={sentVotesLoading || isBanned}>{t('Vote delegate')}</PrimaryButton>
+        <PrimaryButton disabled={sentVotesLoading || isBanned || !currentAddress}>{t('Vote delegate')}</PrimaryButton>
```
Comment (reyraa) [inline src/modules/dpos/validator/components/delegateProfile/delegateVoteButton.js:31]: I don't get this. Why did you add `!currentAddress`?

File: src/modules/dpos/validator/hooks/useVoteAmountField.js [hunk 15]
```diff
@@ -22,7 +22,7 @@ let loaderTimeout = null;
  */
 const getAmountFeedbackAndError = (value, balance, minValue, inputValue) => {
   const { message: feedback } = validateAmountFormat({
-    value,
+    value: +value,
```
Comment (reyraa) [inline src/modules/dpos/validator/hooks/useVoteAmountField.js:26]: Do you need to `Number` it?

**Review by reyraa — CHANGES_REQUESTED** [review]

File: src/modules/dpos/validator/hooks/useVoteAmountField.js (hunk 15 above)
Comment (eniolam1000752) [reply src/modules/dpos/validator/hooks/useVoteAmountField.js:26]: yhea its needed because the validator performs wrongly if value is a string

File: src/modules/dpos/validator/components/delegateProfile/delegateVoteButton.js (hunk 14 above)
Comment (eniolam1000752) [reply src/modules/dpos/validator/components/delegateProfile/delegateVoteButton.js:31]: this was so that when an account isn't selected, the `Vote delegate` button is disabled

File: src/modules/dpos/validator/api/index.test.js (hunk 4 above)
Comment (eniolam1000752) [reply src/modules/dpos/validator/api/index.test.js:114]: hmmm okay. 
Though this happens because my prettier formatter which uses the prettier config on the project formats on save.

File: src/modules/dpos/validator/components/EditVote/index.js (hunk 5 above)
Comment (eniolam1000752) [reply src/modules/dpos/validator/components/EditVote/index.js:210]: so `mode` performs the function of telling the component if its to display the edit content or the add content while `isForm` is used to know if it should render the confirmation modal or not (this is regardless if its in edit or add mode)

File: src/modules/dpos/validator/components/VoteSuccessfulModal/VoteSuccessfulModal.css (hunk 8 above)
Comment (eniolam1000752) [reply src/modules/dpos/validator/components/VoteSuccessfulModal/VoteSuccessfulModal.css]: presently there is no mixin that targets the font-size. I would define a new mixin for this then

File: src/modules/dpos/validator/components/VotingQueue/index.js (hunk 13 above)
Comment (eniolam1000752) [reply src/modules/dpos/validator/components/VotingQueue/index.js]: yhea added it to the comment

**Review by reyraa — APPROVED** [review]
